### PR TITLE
feat: First-token logit gating to suppress chat-residue openers

### LINKED ISSUE #24: Logit gating on first generated token
Chat residue ("Sure,", "Here's", leading newline, leading `I `) shows up at the first token far more than anywhere else. A small per-tokenizer deny list applied as a `-inf` logit mask on the first sampled token eliminates these reliably — cheaper than prompt engineering and model-agnostic.

**Acceptance**
- [ ] Mask applies only to the first generated token.
- [ ] Deny list is configurable per model.
- [ ] Debug mode logs which token was suppressed when the gate fires.
- [ ] Settings toggle, default on.

Parent: #13

### PR DESCRIPTION
**Pull request by Jam-Cai**
Implements issue #24. Instruction-tuned models routinely begin inline-autocomplete suggestions with conversational openers ("Sure,", "Here's", "Of course", a leading newline) — chat-reply behavior leaking into a continuation context. This adds a per-tokenizer deny list applied as a `-inf` logit mask on the first sampled token only, eliminating the residue deterministically without prompt engineering.

## What's in here

**Mechanism**
- `tabby/Support/FirstTokenDenyList.swift` — declarative deny strings. Pure data; no runtime deps.
- `LlamaRuntimeCore` resolves strings to token IDs once per model load against the loaded vocabulary, deduplicating leading-token collisions.
- A second sampler chain prepends `llama_sampler_init_logit_bias` with `-inf` entries and is used **only at `tokenIndex == 0`**; subsequent tokens use the standard sampler so deny-listed tokens can still appear naturally mid-suggestion (e.g. "I" as a second word).

**Conservative starting list**
`"Sure"`, `"Here"`, `"Of course"`, `"Certainly"`, `"\n"`. Every entry is a token that has essentially no legitimate use as the *first* token of a continuation. Notably **not** included despite being suggestive in the issue: `"I "` (high false-positive rate in real prose), `"Let me"` / `"Let"` (too broad), language-specific openers — none of these have measurement to back them. The per-model switch is kept as the extension point so the list can grow from gate-fire log evidence rather than intuition.

**Settings toggle**
"Suppress Chat Openers", default on, persisted under `tabbyFirstTokenGatingEnabled`. Apple Intelligence does not expose logit-level control, so the toggle only renders for the Open Source engine.

**Debug logging** (acceptance criterion)
Subsystem `app.tabby`, category `first-token-gate`. Logs the resolved deny list at model load, plus a per-fire signal whenever the **un-gated argmax** of the raw logits at position 0 is in the deny set. Argmax is checked rather than the sampled token because the sampled token passes through stochastic top-p / min-p stages — argmax precisely answers "did the gate prevent the model's strongest preference from being chosen?".

```
log stream --predicate 'subsystem == "app.tabby" AND category == "first-token-gate"' --level debug
```

## Validation

Ran locally:
- `swiftlint --reporter github-actions-logging` — no new warnings on changed files
- `xcodebuild ... build CODE_SIGNING_ALLOWED=NO` — **BUILD SUCCEEDED**
- `xcodebuild test ... CODE_SIGNING_ALLOWED=NO` — **TEST SUCCEEDED**

`LlamaPromptRendererTests` updated to pass the new `isFirstTokenGatingEnabled` argument to its `SuggestionRequest` constructor.

**Screenshot of the new Settings toggle:** TODO — local build couldn't capture due to Screen Recording permissions; will attach when running on a machine with the permission, or happy to ship without if you'd prefer to verify in your build.

## Linked issues

Closes #24.

## Risk / rollout notes

- Default-on is intentional: the gate only applies a `-inf` mask to a small set of tokens whose argmax-occurrence rate at position 0 is exactly the misbehavior we want to fix. False-positive cost on the conservative starting list should be near zero.
- The gate is a pure sampling-time addition; if it ever needs to be turned off in the field, the toggle disables the second sampler chain and falls back to today's behavior.
- Resolution is per-model-load, so adding entries to the deny list later is a no-op until the next model load.

### CODE REVIEW

File: tabby/Support/FirstTokenDenyList.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    static func denyStrings(for modelFilename: String) -> [String] {
+        switch modelFilename {
+        case "gemma-3-1b-it-Q4_K_M.gguf",
+             "gemma-3n-E4B-it-Q4_K_M.gguf",
+             "Qwen3-0.6B-Q4_K_M.gguf":
+            return Self.conservativeDenyStrings
+        default:
+            return Self.conservativeDenyStrings
+        }
+    }
```
Comment (FuJacob) [inline tabby/Support/FirstTokenDenyList.swift:60]: all case return same deny string list, do we need this then?

**Comment by FuJacob** [comment]
Closing as outdated/superseded. If this work is still needed, please rebuild it from current main.
